### PR TITLE
Client Structure: includes angular material and lazy load

### PR DESCRIPTION
**Pull request by dach020**

### CODE REVIEW

**Comment by dach020** [comment]
Hi. I'd love to do that. But unfortunately, Webpack 4.0 (which we are currently using right now) doesn't support Stylus just yet. https://github.com/shama/stylus-loader/issues/189

Your thoughts?

File: package.json [hunk 1]
```diff
@@ -9,7 +9,8 @@
     "build:server": "rimraf ./dist/server && tsc --project ./src/server",
     "build:client": "rimraf ./dist/client && webpack --progress --colors",
     "e2e:test": "cross-env NODE_ENV=test npm run build:server && jest ./dist/server/app/modules/**/*.controller.test.js",
-    "start": "node ./dist/server/main.js"
+    "start": "node ./dist/server/main.js",
+    "watch": "webpack --progress --watch"
```
Comment (cedrickmandocdoc) [inline package.json]: make this watch:client

**Comment by cedrickmandocdoc** [comment]
Styles: For me, either stylus or sass is ok.

Syntax: We should use same ground on style guide and I choose this one for the client https://github.com/johnpapa/angular-styleguide/blob/master/a2/README.md

Structure: The structure for me is good. using shared modules is good. Just rename the common folder to shared. Then inside it has components, pipes and services maybe.
Somewhere near in the future we will probably add core module which generally its role is to provide something for the app that is general to all. This should be imported only once and unlike the shared module that is being imported whenever it is need.

File: src/client/app/common/shared.module.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    ReactiveFormsModule,
+    SafeHtmlPipe,
+    MaterialModule
+  ],
+  entryComponents: [],
+  schemas: [ CUSTOM_ELEMENTS_SCHEMA, NO_ERRORS_SCHEMA ]
+})
+
+export class SharedModule {
+  public static forRoot(): ModuleWithProviders {
```
Comment (cedrickmandocdoc) [inline src/client/app/common/shared.module.ts]: I am not sure why we need this method?

File: src/client/app/marketing/marketing.module.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { CUSTOM_ELEMENTS_SCHEMA, NgModule } from '@angular/core';
```
Comment (cedrickmandocdoc) [inline src/client/app/marketing/marketing.module.ts]: we will not support custom elements schema, to be strict what tags we are using

File: src/client/app/material.module.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
+import { OverlayModule } from '@angular/cdk/overlay';
```
Comment (cedrickmandocdoc) [inline src/client/app/material.module.ts]: I've been thinking if we need this material module? should we put this in shared? I am not sure though or maybe just put this inside "material" folder, what do you think?

File: src/client/index.html [hunk 5]
```diff
@@ -4,6 +4,7 @@
   <meta charset="UTF-8">
   <meta name="viewport" content="width=device-width, initial-scale=1.0">
   <meta http-equiv="X-UA-Compatible" content="ie=edge">
+  <link href="https://fonts.googleapis.com/css?family=Roboto:100,300,400,700" rel="stylesheet">
```
Comment (cedrickmandocdoc) [inline src/client/index.html]: this should be imported on our main style. styles.scss

File: src/client/app/material.module.ts (hunk 4 above)
Comment (jmaicaaan) [reply src/client/app/material.module.ts]: Just keep it inside the material module. Doesn't need to be extracted from there. It's fine

**Pull request merged**